core/src/main/java/org/apache/iceberg/UpdateSnapshotReferencesOperation.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Map;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+class UpdateSnapshotReferencesOperation implements PendingUpdate<Map<String, SnapshotRef>> {
+
+  private final TableOperations ops;
+  private final Map<String, SnapshotRef> updatedRefs;
+  private TableMetadata base;
+
+  UpdateSnapshotReferencesOperation(TableOperations ops) {
+    this.ops = ops;
+    this.base = ops.current();
+    this.updatedRefs = Maps.newHashMap(base.refs());
+  }
+
+  @Override
+  public Map<String, SnapshotRef> apply() {
+    return refDiff(base.refs(), updatedRefs);
```

Comment:
I think this should return `updatedRefs` instead of the changed refs. I think there is more value in returning to the caller the final state, rather than changes.